test/test_eager_transforms.py

```diff
@@ -2134,6 +2134,56 @@ def compute_loss(weights, image, target):
         self.assertEqual(result_grads, expected_grads)
 
 
+class TestFunctionalize(TestCase):
+    def _check_functionalize_correctness(self, f: Callable, inpt: torch.Tensor) -> None:
+        inpt1 = inpt.clone()
+        inpt2 = inpt.clone()
+
+        expected_outputs = f(inpt1)
+        actual_outputs = vmap(functionalize(f))(inpt2.unsqueeze(0))[0].squeeze()
+        # Check that outputs are the same
+        self.assertEqual(actual_outputs, expected_outputs)
```

Comment:
Lol functionalize(f) could just return f and these tests wouldn't know better :D.

This seems fine for now though -- we'll probably find some better way of testing this in the future.

Reply:
Yep and agreed :D

Long term, I think a combination of OpInfo tests in core + loggingtensor expecttests in functorch would be a lot better.